pom.xml

```diff
@@ -1852,6 +1852,36 @@
           </validations>
         </configuration>
       </plugin>
+
+      <plugin>
+        <groupId>dev.dimlight</groupId>
+        <artifactId>shellcheck-maven-plugin</artifactId>
```

Comment:
Can we use shell tool directly ?
It is weird to use shell tool through maven

Reply:
As far as I have understood, if we want shellcheck to run locally we can do that by:

1. Terminal: `shellcheck yourscript`

2.  Maven Plugin or Dependency.

Reply:
We need option 1. We can not use maven plugin to avoid dependency to shell tool during build. She'll files are not value files for our users, they are infrastructure helpers for CI.

Reply:
So, it means we don't have to integrate shellcheck. Do we want to use the terminal to find out the violations in a particular .sh file?

Reply:
We do not want spellcheck in maven build but we want it in CI, as just another shell execution job, we have numerous already.
For example https://github.com/checkstyle/checkstyle/blob/master/.ci/checkchmod.sh
This script also care about best practices of shell files :).

Reply:
We need to install it in Travis by apt, if not already installed.
On local we will install it by apt or similar package managers.

Reply:
Shellcheck integration with Travis CI is done. Please review

Reply:
@Rahulkhinchi03 Travis did not run on last commit. Please temporarily remove ignore file and show that your new job will fail CI.

Reply:
@nick-mancuso What is the reason for Travis not running. I tried to find the reason and on Travis, it was written "abuse deducted"

Reply:
@Rahulkhinchi03 please share a link to where you see this, always share a link

Reply:
https://app.travis-ci.com/github/checkstyle/checkstyle/requests

Reply:
I am not familiar with this, please investigate and share results here. I suspect this PR may be getting flagged for a bunch of changes with shell commands (security concern)

Reply:
https://github.com/travis-ci/travis-ci/issues/8626#issuecomment-339270186 By This Comment, I removed Travis ci from my personal account and I believe that checkstyle needs to regrant Travis-Ci access to checkstyle?

Reply:
@romani The Travis CI is not running on my PR. I believe that it is due to my PR issue or something? Please let me know if it is a from Travis or if it is from my side, I may create a new PR